cpp/util/SFURL.cpp

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#include "snowflake/SFURL.hpp"
+#include "snowflake/SF_CRTFunctionSafe.h"
+
+namespace Snowflake
+{
+namespace Client
+{
+
+/// Functions of QueryParams
+SFURL::QueryParams::QueryParams(const SFURL::QueryParams &params, bool &cacheValid, std::string &cacheURL)
+:m_cacheValid(&cacheValid), m_cacheURL(&cacheURL)
+{
+  m_list = params.m_list;
+  for (auto iter = m_list.begin(); iter != m_list.end(); iter++)
```

Comment:
let's add braces even when the body is a single line - here and in other places

Reply:
checked and done all other places.